src/app/loadout-builder/LoadoutBuilder.tsx

```diff
@@ -338,9 +339,14 @@ function LoadoutBuilder({
           ReactDOM.createPortal(
             <ModPicker
               classType={selectedStore.classType}
-              lockedMods={lockedMods}
+              lockedMods={_.mapValues(lockedMods, (mods) => mods?.map((mod) => mod.modDef))}
```

Comment:
Not sure if the mod picker is using `connect`, but it's worth noting that these two changed lines (`lockedMods` and `onAccept`) will force the ModPicker to render every time LoadoutBuilder re-renders. Consider `useMemo` and `useCallback` if that isn't what you want. That said, if this was going to re-render every time LoadoutBuilder re-rendered anyway, then it's maybe not worth it.

Reply:
It will rerender everytime the loadout builder rerenders while mod picker is open, the line above this is missing but is

```ts
{modPicker.open &&
          ReactDOM.createPortal(
            <ModPicker
              classType={selectedStore.classType}
              lockedMods={_.mapValues(lockedMods, (mods) => mods?.map((mod) => mod.modDef))}
              initialQuery={modPicker.initialQuery}
              onAccept={(newLockedMods: PluggableItemsByPlugCategoryHash) =>
                lbDispatch({
                  type: 'lockedModsChanged',
                  lockedMods: newLockedMods,
                })
              }
              onClose={() => lbDispatch({ type: 'closeModPicker' })}
            />,
            document.body
          )}
```

I figure it's not being interacted with so it shouldn't reload. If state is updated at the app level and forces a reload we will want to rerender anyways. Can you think of a condition outside of this that we would care about?

Reply:
Seems OK.